llm_behavior_eval/evaluation_utils/util_functions.py

```diff
@@ -38,11 +48,50 @@ def safe_apply_chat_template(
             messages[0]["content"] = f"{sys_msg}\n\n{messages[0]['content']}"
         else:
             messages.insert(0, {"role": "user", "content": sys_msg})
-    return str(
-        tokenizer.apply_chat_template(
-            messages, tokenize=False, add_generation_prompt=True
+
+    # Choose formatting based on whether the model is multimodal
+    def _apply_chat_template(
+        messages_like: list[dict[str, Any]] | list[dict[str, str]],
+    ):
+        """
+        Call tokenizer.apply_chat_template while remaining compatible with
+        tokenizers that don't support the `reasoning` kwarg.
+        """
+        try:
+            return tokenizer.apply_chat_template(
+                messages_like,
+                tokenize=False,
+                add_generation_prompt=True,
+                reasoning=False,
+            )
+        except TypeError:
+            # Older or stub tokenizers may not accept `reasoning`
+            return tokenizer.apply_chat_template(
+                messages_like,
+                tokenize=False,
+                add_generation_prompt=True,
+            )
+
+    if is_multimodal:
+        # Multimodal: list-of-parts with type "text"
+        chat_messages_mm: list[dict[str, Any]] = []
```

Comment:
```suggestion
        multimodal_chat_messages: list[dict[str, Any]] = []
```
Please don't use non-common shorthands

Reply:
ok thanks